cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -640,7 +645,7 @@ public String getAncillaryConfigMapName() {
      *
      * @return null
      */
-    protected List<EnvVar> getEnvVars() {
+    protected List<EnvVar> getEnvVars(Reconciliation reconciliation) {
```

Comment:
Why did you opted for passing the reconciliation object around everywhere? Why not set it once in constructor? It would make the code look much nicer and easier to maintain.